### PR TITLE
Fixes over triggering of binary logic

### LINKED ISSUE #1403: BinaryLogic: Over triggering, throwing errors
Generic queries, like "what do number and letter codes in a lens name mean" are triggering this IA and throwing errors. We need to fix this asap!

Also, let’s remove the regex trigger in favour of a much more efficient words trigger:

```perl
triggers any => "and", "or", "xor", "⊕", "∧", "∨","not", "¬";
```

We can move the regex's into the `handle` function and use them to verify our inputs. The regexs need to be improved to prevent unexpected inputs and the captures from the regex need to be checked to ensure they are valid numbers.

/cc @Sloff

### PR DESCRIPTION
**Pull request by Sloff**
Hi @moollaza this will fix #1403

### CODE REVIEW

**Comment by moollaza** [comment]
@Sloff wow, you *just* beat me. I was about to submit a PR where I've moved the subtitle building loop below the parser runs. That way we catch errors and bail before formatting the subtitle.

**Comment by moollaza** [comment]
I also added a bunch of failing tests. I think we should add in more "failing" tests to verify queries that could trigger but shouldn't return an answer, don't return anything:

https://github.com/duckduckgo/zeroclickinfo-goodies/compare/zaahir/fix-binary-logic

File: lib/DDG/Goodie/BinaryLogic.pm [hunk 1]
```diff
@@ -99,14 +99,18 @@ sub BinaryLogic_Actions::do_not {
 }
 
 handle query_raw => sub {
+    my $input = $_;
+    
+    my $testError = $input;
+    $testError =~ s/(?:0x|0b|[\d\s]|and|or|xor|not|\(|\)|⊕|∧|∨|¬)//ig;
+    return undef if length $testError != 0;
```
Comment (moollaza) [inline lib/DDG/Goodie/BinaryLogic.pm]: @Sloff you actually just want to `return` here, not `return undef`

**Comment by moollaza** [comment]
@Sloff I've combined our PR's here: https://github.com/duckduckgo/zeroclickinfo-goodies/pull/1405

Thanks a ***lot*** for the very fast response.

**Comment by Sloff** [comment]
Pleasure @moollaza!

**Pull request closed without merging**